reactor-core/src/main/java/reactor/core/publisher/EmitterProcessor.java

```diff
@@ -192,6 +205,39 @@ public Context currentContext() {
 		return Operators.multiSubscribersContext(subscribers);
 	}
 
+
+	private boolean isDetached() {
+		return s == Operators.cancelledSubscription() && done && error instanceof CancellationException;
+	}
+
+	private boolean detach() {
+		if (Operators.terminate(S, this)) {
+			done = true;
+			CancellationException detachException = new CancellationException("the ManyWithUpstream sink had a Subscription to an upstream which has been manually cancelled");
+			if (ERROR.compareAndSet(EmitterProcessor.this, null, detachException)) {
+				Queue<T> q = queue;
+				if (q != null) {
+					q.clear();
+				}
+				for (FluxPublish.PubSubInner<T> inner : terminate()) {
+					inner.actual.onError(detachException);
+				}
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public Disposable subscribeTo(Publisher<? extends T> upstream) {
+		EmitterDisposable ed = new EmitterDisposable(this);
+		if (UPSTREAM_DISPOSABLE.compareAndSet(this, null, ed)) {
+			upstream.subscribe(this);
+			return ed;
+		}
+		return Disposables.disposed();
```

Comment:
should not we throw an exception instead? if we return disposed disposable, the a caller will never know if it was a successful subscription